### PR TITLE
fixed multiple line bug (#37)

### LINKED ISSUE #37: StanModule errors if file contains more than 1 line
Not tested this yet but based on a code review we believe the StanModule object will error if it tries to load a Stan file that contains more than 1 line of code.

This error should be fixed and a unit test put in place to show that it is fixed.

### PR DESCRIPTION
**Pull request by gowerc**
Closes #37

### CODE REVIEW

File: R/stan_module.R [hunk 1]
```diff
@@ -31,9 +32,9 @@ StanModule <- setClass(
 read_stan <- function(string) {
     system_file <- system.file("stanparts", string, package = "jmpost")
     if (file.exists(string)) {
-        out <- readLines(string)
+        out <- read_file(string)
     } else if (file.exists(system_file)) {
```
Comment (G-Kazantzidis) [inline R/stan_module.R:36]: @gowerc is it possible to replace the `file.exists(system_file)` with `is_file(system_file)`. `is_file` is a function from package `fs` and explicitly checks whether a file exists and not whether the provided path is TRUE. I created a relevant issue [here ](https://github.com/Genentech/jmpost/issues/39).

**Comment by gowerc** [comment]
@G-Kazantzidis I added the code here to resolve #39  as well apologies I got confused on wether you were doing it or if you were asking me to address it but it felt related so I threw it in here

File: R/stan_module.R [hunk 2]
```diff
@@ -23,17 +23,18 @@ StanModule <- setClass(
 )
 
 
+
 #' read_stan returns stan code as a character.
 #'
 #' @param file Character, either the absolute path of a stan file, or the name of the stan
 #' file in the package directory or the stan code as a string.
```
Comment (G-Kazantzidis) [inline R/stan_module.R:30]: #' @importFrom fs is_file

File: R/stan_module.R [hunk 3]
```diff
@@ -212,3 +213,37 @@ remove_blank_strings <- function(x) {
...
+
+#' Read entire file as a single string
+#'
+#' Simple utility function to read in a file as 1
+#' continous string
+#'
+#' @param filename Location of file to read in
+#'
+read_file <- function(filename) {
+    paste0(readLines(filename), collapse = "\n")
```
Comment (G-Kazantzidis) [inline R/stan_module.R:225]: I am thinking that maybe changing line in the end of the character string could be helpful for later `paste0(paste0(readLines(filename), collapse = "\n"), "\n")`. eg when we will add "}", we want them to be in an empty line at the end.

File: R/stan_module.R (hunk 2 above)
Comment (G-Kazantzidis) [reply R/stan_module.R:30]: apologies, I just saw that you create a new `is_file` function below

File: R/stan_module.R (hunk 3 above)
Comment (gowerc) [reply R/stan_module.R:225]: I think that would make more sense to go into the `as.character()` method no ?  As in (though unlikely) there is a chance this function could get used for other purposes where it doesn't make sense to have that trailing newline.

Comment (G-Kazantzidis) [reply R/stan_module.R:225]: Your suggestion sounds more reasonable way to proceed.

**Review by G-Kazantzidis — APPROVED** [review]

**Pull request merged**